metrics/src/main/java/tech/pegasys/pantheon/metrics/MetricsSystem.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.metrics;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+public interface MetricsSystem {
+
+  default Counter createCounter(
```

Comment:
Why a default method?  Doesn't the var-args already handle the zero arg case?

Reply:
This returns a raw `Counter` the one with varargs returns `LabelledMetric<Counter>` so you have to supply label values before you can call `inc()`.

Reply:
Got it, however createLabeledCounter for the varargs would look better in calling code where return value isn't always obvious.

Reply:
True, changed.